tests/auto/jobs/build.py

```diff
@@ -152,68 +198,101 @@ def clone_pr_repo(job_obj, workdir):
-                bldate = bldate.strip(' ')
-                logger.info(f'bldate is "{bldate}"')
-                logger.info(f'Type bldate: {type(bldate)}')
-                bl_format = '%Y%m%d'
-                try:
-                    datetime.datetime.strptime(bldate, '%Y%m%d')
-                except ValueError:
-                    logger.info(f'Date {bldate} is not formatted YYYYMMDD')
-                    raise ValueError
-    if not BLDATEFOUND:
-        job_obj.comment_text_append('BL_DATE not found in rt.sh.'
-                                    'Please manually edit rt.sh '
-                                    'with BL_DATE={bldate}')
-        job_obj.job_failed(logger, 'get_bl_date()')
-    logger.info('Finished get_bl_date')
-
-    return bldate
+    logger.info('Log file was processed')
+    logger.info(f'Status of build: {logfile_pass}')
+
+    return logfile_pass
 
 
 def process_logfile(job_obj, ci_log):
+    """
+    Runs after code has been cloned and built
+    Checks to see whether build was successful or failed
+    """
     logger = logging.getLogger('BUILD/PROCESS_LOGFILE')
     fail_string = 'FAIL'
     build_failed = False
     if os.path.exists(ci_log):
-        with open(ci_log) as f:
-            for line in f:
+        with open(ci_log) as fname:
+            for line in fname:
                 if fail_string in line:
                     build_failed = True
-                    job_obj.comment_text_append(f'{line.rstrip(chr(10))}')
+                    job_obj.comment_append(f'{line.rstrip()}')
         if build_failed:
-            job_obj.job_failed(logger, f'{job_obj.preq_dict["action"]}')
-            logger.info('Build failed') 
+            job_obj.send_comment_text()
+            logger.info('Build failed')
         else:
-            logger.info('Build was successful') 
-        return not build_failed 
+            logger.info('Build was successful')
+        return not build_failed
     else:
         logger.critical(f'Could not find {job_obj.machine}'
                         f'.{job_obj.compiler} '
                         f'{job_obj.preq_dict["action"]} log')
         raise FileNotFoundError
+
+
+def process_gen(job_obj, gen_log_loc, gen_log_name):
+    """
+    Runs after a rocoto workflow has been generated
+    Checks to see if an error has occurred
+    """
+    logger = logging.getLogger('BUILD/PROCESS_GEN')
+    gen_log = f'{gen_log_loc}/{gen_log_name}'
+    error_string = 'ERROR'
+    gen_failed = False
+    if os.path.exists(gen_log):
+        with open(gen_log) as fname:
+            for line in fname:
+                if error_string in line:
+                    job_obj.comment_append('Generating Workflow Failed')
+                    gen_failed = True
+                    logger.info('Generating workflow failed')
+                if gen_failed:
+                    job_obj.comment_append(f'{line.rstrip()}')
+
+
+def process_expt(job_obj, expt_dirs):
```

Comment:
`expt_dirs` here implies a list, but then is used as a string below. Does it make sense to make this variable name singular?

Reply:
This is the literal name of the subdirectory used by the workflow test shell scripts. It can contain one or more experiment directories.

Reply:
Ahhh! Okay. I see what's happening on line 278. Would it be clearer to use something like `expts_base_dir`? Apologies for being picky. I usually just process plural variables as a collection of items.

Reply:
I changed it on my laptop. I'll push it up after a successful test.